### PR TITLE
Add `MockSegmentationInterface` 

### PR DESCRIPTION
**Pull request by h-mayorquin**
As in the title.

### CODE REVIEW

File: src/neuroconv/tools/testing/mock_interfaces.py [hunk 1]
```diff
@@ -162,24 +165,52 @@ class MockImagingInterface(BaseImagingExtractorInterface):
     A mock imaging interface for testing purposes.
     """
 
+    ExtractorModuleName = "roiextractors.testing"
```
Comment (pauladkisson) [inline src/neuroconv/tools/testing/mock_interfaces.py:168]: What do these changes have to do with a MockSegmentationInterface?

File: src/neuroconv/tools/testing/mock_interfaces.py [hunk 2]
```diff
@@ -162,24 +165,52 @@ class MockImagingInterface(BaseImagingExtractorInterface):
...
+        dtype : str, optional
+            The data type of the generated imaging data (e.g., 'uint16'), by default 'uint16'.
+        verbose : bool, optional
+            If True, prints additional information during processing, by default True.
+        seed : int, optional
+            Random seed for reproducibility, by default 0.
+        photon_series_type : Literal["OnePhotonSeries", "TwoPhotonSeries"], optional
+            The type of photon series for the mock imaging data, either "OnePhotonSeries" or
+            "TwoPhotonSeries", by default "TwoPhotonSeries".
+        verbose : bool, default True
```
Comment (pauladkisson) [inline src/neuroconv/tools/testing/mock_interfaces.py:202]:
```suggestion
        verbose : bool, default False
```

File: src/neuroconv/tools/testing/mock_interfaces.py [hunk 3]
```diff
@@ -190,3 +221,74 @@ def get_metadata(self, photon_series_type: Optional[Literal["OnePhotonSeries", "
...
+        has_dff_signal : bool, optional
+            whether a relative (df/f) fluorescence signal is desired in the object, by default True.
+        has_deconvolved_signal : bool, optional
+            whether a deconvolved signal is desired in the object, by default True.
+        has_neuropil_signal : bool, optional
+            whether a neuropil signal is desired in the object, by default True.
+        seed: int, default 0
+            seed for the random number generator, by default 0
+        verbose : bool, optional
+            controls verbosity, by default True.
```
Comment (pauladkisson) [inline src/neuroconv/tools/testing/mock_interfaces.py:273]:
```suggestion
            controls verbosity, by default False.
```

File: src/neuroconv/tools/testing/mock_interfaces.py [hunk 4]
```diff
@@ -162,24 +165,52 @@ class MockImagingInterface(BaseImagingExtractorInterface):
...
+            The number of frames in the mock imaging data, by default 30.
+        num_rows : int, optional
+            The number of rows (height) in each frame of the mock imaging data, by default 10.
+        num_columns : int, optional
+            The number of columns (width) in each frame of the mock imaging data, by default 10.
+        sampling_frequency : float, optional
+            The sampling frequency of the mock imaging data in Hz, by default 30.
+        dtype : str, optional
+            The data type of the generated imaging data (e.g., 'uint16'), by default 'uint16'.
+        verbose : bool, optional
```
Comment (pauladkisson) [inline src/neuroconv/tools/testing/mock_interfaces.py:195]: One of these verbose's is redundant.

**Review by pauladkisson — CHANGES_REQUESTED** [review]

File: src/neuroconv/tools/testing/mock_interfaces.py (hunk 1 above)
Comment (h-mayorquin) [reply src/neuroconv/tools/testing/mock_interfaces.py:168]: This is related to make the MockImagingInterface to work with some of the magic in the tests mixings. I can make this changes in a different PR if that is what you are asking.

File: src/neuroconv/tools/testing/mock_interfaces.py (hunk 4 above)
Comment (h-mayorquin) [reply src/neuroconv/tools/testing/mock_interfaces.py:195]: Thanks!

File: src/neuroconv/tools/testing/mock_interfaces.py (hunk 1 above)
Comment (h-mayorquin) [reply src/neuroconv/tools/testing/mock_interfaces.py:168]: Yeah, I think this is a good idea anyways. Check:

#1076

**Comment by h-mayorquin** [comment]
OK, did the changes, #1076 should come before.

**Review by weiglszonja — APPROVED** [review]
LGTM

**Review by pauladkisson — APPROVED** [review]

**Pull request merged**